### PR TITLE
Add same_anchor util.

### PR DESCRIPTION
**Pull request by ueshin**
Adding `same_anchor` util to check if the anchors of the given DataFrame or Series are the same or not.

### CODE REVIEW

File: databricks/koalas/series.py [hunk 1]
```diff
@@ -60,9 +60,10 @@
 from databricks.koalas.ml import corr
 from databricks.koalas.utils import (
     validate_arguments_and_invoke_function,
-    scol_for,
     combine_frames,
     name_like_string,
+    same_anchor,
+    scol_for,
```
Comment (itholic) [inline databricks/koalas/series.py:66]: It's really not a big deal, but I'm just curious that why the position of `scol_for` has been moved?

**Comment by itholic** [comment]
LGTM

File: databricks/koalas/utils.py [hunk 2]
```diff
@@ -35,25 +35,27 @@
...
         this_kdf = this
     else:
-        assert isinstance(this, Series), type(this)
+        assert isinstance(this, IndexOpsMixin), type(this)
         this_kdf = this._kdf
     if isinstance(that, DataFrame):
         that_kdf = that
     else:
-        assert isinstance(that, Series), type(that)
+        assert isinstance(that, IndexOpsMixin), type(that)
```
Comment (itholic) [inline databricks/koalas/utils.py:58]: Looks better!

File: databricks/koalas/series.py (hunk 1 above)
Comment (ueshin) [reply databricks/koalas/series.py:66]: The order should basically be alphabetical order.
I'd like black or something to manage the order..

Comment (itholic) [reply databricks/koalas/series.py:65]: Ah, I got it! Sorry for annoying and thanks for the reply !

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**